### PR TITLE
enh(platform): Use API to select metrics in virtual metrics configuration form

### PR DESCRIPTION
**Pull request by hyahiaoui-ext**
## Description

Replacing www/include/views/graphs/common/makeXML_ListMetrics.php by the select2 for metrics

Listed metrics are from host/service couple selected in form

Previews: 

Before:
image
After:
image

**Fixes** # MON-14359

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [x] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [x] 21.04.x
- [x] 21.10.x
- [x] 22.04.x
- [x] 22.10.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Comment by hyahiaoui-ext** [comment]
Hey dear reviewers 👋, i know i've made changes in the composer.json and lock files, just to test the build, i'll revert them after mergin the branch in "test-lib"!

File: www/include/views/componentTemplates/formComponentTemplate.ihtml [hunk 1]
```diff
@@ -34,6 +34,9 @@
 			{if $o == "a" || $o == "c"}
 				<td id="td_list_metrics" style="padding: 0px; border-width: 0px;">
 					&nbsp;<img src="./img/icons/arrow-left.png" class="ico-14" onclick="insertValueQuery()" />&nbsp;
+				    <td class="log-select" style="padding-bottom: 4px">
+					    <select id="sl_list_metrics" size="1"></select>
+				    </td>
```
Comment (callapa) [inline www/include/views/componentTemplates/formComponentTemplate.ihtml]: You should not include a column (td) in an existing column.

File: www/include/views/virtualMetrics/formVirtualMetrics.ihtml [hunk 2]
```diff
@@ -49,6 +49,9 @@
 			{if $o == "a" || $o == "c"}
 				<td id="td_list_metrics" style="padding: 0px; border-width: 0px;">
 					&nbsp;<img src="./img/icons/arrow-left.png" class="ico-14" onclick="insertValueQuery()" />&nbsp;
+					<td class="log-select" style="padding-bottom: 4px">
+						 <select id="sl_list_metrics" size="1"></select>
+					</td>
```
Comment (callapa) [inline www/include/views/virtualMetrics/formVirtualMetrics.ihtml]: You should not include a column (td) in an existing column.

**Review by callapa — CHANGES_REQUESTED** [review]

File: www/include/views/componentTemplates/formComponentTemplate.ihtml (hunk 1 above)
Comment (hyahiaoui-ext) [reply www/include/views/componentTemplates/formComponentTemplate.ihtml]: @callapa done :)

**Review by callapa — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Pull request merged**